File: src/main/java/edu/cit/estillore/MentorMatch/MentorMatch/repository/AvailabilityRepository.java
package edu.cit.estillore.MentorMatch.MentorMatch.repository;

import edu.cit.estillore.MentorMatch.MentorMatch.entities.Availability;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface AvailabilityRepository extends JpaRepository<Availability, Long> {
    List<Availability> findByTutorProfileId(Long tutorProfileId);
    void deleteByTutorProfileId(Long tutorProfileId);
}

File: android/app/src/main/java/edu/cit/estillore/mentormatch/data/repository/AvailabilityRepository.kt
package edu.cit.estillore.mentormatch.data.repository

import edu.cit.estillore.mentormatch.data.api.AvailabilityApi
import edu.cit.estillore.mentormatch.data.api.ErrorParser
import edu.cit.estillore.mentormatch.data.model.AvailabilityRequest
import edu.cit.estillore.mentormatch.data.model.AvailabilitySlot

class AvailabilityRepository(private val availabilityApi: AvailabilityApi) {

    suspend fun forTutorProfile(tutorProfileId: Long): Result<List<AvailabilitySlot>> = runCatching {
        val response = availabilityApi.forTutorProfile(tutorProfileId)
        if (response.isSuccessful) response.body() ?: emptyList()
        else throw IllegalStateException(ErrorParser.parse(response))
    }

    suspend fun replaceMine(slots: List<AvailabilitySlot>): Result<List<AvailabilitySlot>> = runCatching {
        val response = availabilityApi.replaceMine(AvailabilityRequest(slots))
        if (response.isSuccessful) response.body() ?: emptyList()
        else throw IllegalStateException(ErrorParser.parse(response))
    }
}